tests/prefetch_related/tests.py

```diff
@@ -1161,3 +1161,84 @@ def test_bug(self):
         prefetcher = get_prefetcher(self.rooms[0], 'house')[0]
         queryset = prefetcher.get_prefetch_queryset(list(Room.objects.all()))[0]
         self.assertNotIn(' JOIN ', force_text(queryset.query))
+
+
+class UUIDPrefetchRelated(TestCase):
+
+    def test_prefetch_related_from_uuid_model(self):
+        Pet.objects.create(name='Fifi').people.add(
+            Person.objects.create(name='Ellen'),
+            Person.objects.create(name='George')
+        )
+
+        with self.assertNumQueries(2):
+            pet = Pet.objects.prefetch_related('people').get(name='Fifi')
+        with self.assertNumQueries(0):
+            self.assertEqual(2, len(pet.people.all()))
+
+    def test_prefetch_related_to_uuid_model(self):
+        Person.objects.create(name='Bella').pets.add(
+            Pet.objects.create(name='Socks'),
+            Pet.objects.create(name='Coffee')
+        )
+
+        with self.assertNumQueries(2):
+            person = Person.objects.prefetch_related('pets').get(name='Bella')
+        with self.assertNumQueries(0):
+            self.assertEqual(2, len(person.pets.all()))
+
+    def test_prefetch_related_from_uuid_model_to_uuid_model(self):
+        fleas = [Flea.objects.create() for i in range(3)]
+        Pet.objects.create(name='Fifi').fleas_hosted.add(*fleas)
+        Pet.objects.create(name='Bobo').fleas_hosted.add(*fleas)
+
+        with self.assertNumQueries(2):
+            pet = Pet.objects.prefetch_related('fleas_hosted').get(name='Fifi')
+        with self.assertNumQueries(0):
+            self.assertEqual(3, len(pet.fleas_hosted.all()))
+
+        with self.assertNumQueries(2):
+            flea = Flea.objects.prefetch_related('pets_visited').get(pk=fleas[0].pk)
+        with self.assertNumQueries(0):
+            self.assertEqual(2, len(flea.pets_visited.all()))
+
+    def test_prefetch_related_with_lookups(self):
+        house = House.objects.create(name='Redwood', address='Arcata')
```

Comment:
did you consider using `setUpTestData` for this class? Might make things a bit simpler. In particular, this test is a bit long and I think each `From..., prefetch ....` could be a separate test method.